pkg/certificate/providers/config.go

```diff
@@ -44,7 +43,7 @@ var getCA = func(i certificate.Issuer) (pem.RootCertificate, error) {
 // NewCertificateManager returns a new certificate manager with a MRC compat client.
 // TODO(4713): Remove and use NewCertificateManagerFromMRC
 func NewCertificateManager(ctx context.Context, kubeClient kubernetes.Interface, kubeConfig *rest.Config,
-	providerNamespace string, option Options, computeClient compute.Interface, checkInterval time.Duration, trustDomain string) (*certificate.Manager, error) {
+	providerNamespace string, option Options, computeClient ProvidersInfraClient, checkInterval time.Duration, trustDomain string) (*certificate.Manager, error) {
```

Comment:
providers isn't a very descriptive name. Provider of what? I think InfraClient is good enough on its own?